internal/services/hdinsight/hdinsight_kafka_cluster_resource.go

```diff
@@ -213,23 +208,9 @@ func resourceHDInsightKafkaCluster() *pluginsdk.Resource {
 					"kafka_management_node": SchemaHDInsightNodeDefinition("roles.0.kafka_management_node", hdInsightKafkaClusterKafkaManagementNodeDefinition, false),
 				},
 			},
-			Deprecated: "`kafka_management_node` will be removed in version 4.0 of the AzureRM Provider since it no longer support configurations from the user",
 		}
-	} else {
-		resource.Schema["roles"] = &pluginsdk.Schema{
-			Type:     pluginsdk.TypeList,
-			Required: true,
-			MaxItems: 1,
-			Elem: &pluginsdk.Resource{
-				Schema: map[string]*pluginsdk.Schema{
-					"head_node": SchemaHDInsightNodeDefinition("roles.0.head_node", hdInsightKafkaClusterHeadNodeDefinition, true),
 
-					"worker_node": SchemaHDInsightNodeDefinition("roles.0.worker_node", hdInsightKafkaClusterWorkerNodeDefinition, true),
-
-					"zookeeper_node": SchemaHDInsightNodeDefinition("roles.0.zookeeper_node", hdInsightKafkaClusterZookeeperNodeDefinition, true),
-				},
-			},
-		}
+		resource.Schema["roles"].Elem.(*pluginsdk.Resource).Schema["kafka_management_node"].Elem.(*pluginsdk.Resource).Schema["username"].Deprecated = "`username` will become Computed only in version 4.0 of the AzureRM Provider as the service auto-generates a value for this property"
```

Comment:
Difficult to tell in this diff, but it looks like line 196-211 can be removed since that's now identical to `roles` in the schema further up?

Reply:
the `kafka_management_node` schema changes from `SchemaHDInsightNodeDefinition` to `SchemaHDInsightNodeDefinitionKafka` in 4.0 - the whole `roles` block was processed previously so I kept that the same in the hope it made the change easier to see in the diff (apparently that didn't work out) and to remove later.